traitlets/config/application.py

```diff
@@ -535,13 +535,25 @@ def _load_config_files(cls, basefilename, path=None, log=None, raise_config_file
         if not isinstance(path, list):
             path = [path]
         for path in path[::-1]:
-            # path list is in descending priority order, so load files backwards:
-            pyloader = cls.python_config_loader_class(basefilename+'.py', path=path, log=log)
             if log:
                 log.debug("Looking for %s in %s", basefilename, path)
-            jsonloader = cls.json_config_loader_class(basefilename+'.json', path=path, log=log)
+            # path list is in descending priority order, so load files backwards:
+            def new_loader(name, path):
```

Comment:
No reason to define `new_loader` inside the for-loop over `path[::-1]` on line 537